From patchwork Wed Jul 24 14:39:04 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136358
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tyhH0gbgz9s8m
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:41:03 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id 92844C21D9A; Wed, 24 Jul 2019 14:39:46 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=0.8 required=5.0 tests=UPPERCASE_50_75 autolearn=no
 autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 882B5C21D72;
 Wed, 24 Jul 2019 14:39:40 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 7C7EBC21C27; Wed, 24 Jul 2019 14:39:35 +0000 (UTC)
Received: from mail-wr1-f65.google.com (mail-wr1-f65.google.com
 [209.85.221.65])
 by lists.denx.de (Postfix) with ESMTPS id 2B1D6C21BE5
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:35 +0000 (UTC)
Received: by mail-wr1-f65.google.com with SMTP id x1so32293472wrr.9
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:35 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=8Z82pQdbffuUA8JpsYMwwHCjbZBONEoOwfFh2hjslOs=;
 b=rJPUcmpkh37qDajtL1440DjXD+gxop4QeK685/B+RpJIrTVrBVPe9Z8ahOzM4t6MfA
 R9KYo3KoDOOvp0dm6Ovih0rgEVLjUL9K+/tzqQi9Y3YbflK/VRA7WUuzKWq7o/732qqh
 CLtDHeYCaaAepsl6TzeXSndNd74QoJJFlMtFUf2kDHV3ERIqIDvRd2LI1144GMTFOBjj
 gkZr28fm0BVQPbHVN9p7mZ4vXcKUf2I2Wkje7P6EN6IJTrIcKZvcT2IpnzvsEdbPJ14r
 1JkPI3Bg7pf7iTVAcHDhZ0HWOxnuRuB3KzmC0WkOgYIWpjgUGdNCIys1VHusa8JGUmtz
 X/OQ==
X-Gm-Message-State: APjAAAV4zaznNtIjQJuMF7dLBX4DR9QvnvmUcRzNftXStiuRuXXWnXS1
 MP7oLCr936klnQUkQAkOlyr/cP0s
X-Google-Smtp-Source: APXvYqxJiPSQ9vd5GlT71/+cluS1CRb303VcbpLDq5FS1UdhoChDRrXDGMI6wz5akinycJIMwNQZrg==
X-Received: by 2002:adf:f3d1:: with SMTP id g17mr58097903wrp.38.1563979174441; 
 Wed, 24 Jul 2019 07:39:34 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.33
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:34 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:04 +0100
Message-Id: <20190724143911.29770-2-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 1/8] RPI: Add defconfigs for rpi4 (32/64)
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

This defines a minimum defconfig for each of the two Raspberry Pi 4
variants. One notable difference is that we don't have a embedded dt for
this board given that the fw supplies us with one which we can reuse.
Furthermore, the ram size is not queryable through mbox interface as the
maximum reported size is 1G. The fw patches the dt with the right
memory configuration and uboot uses it as it is. We avoid u-boot
touching this configuration by making sure CONFIG_ARCH_FIXUP_FDT_MEMORY
is deactivated.

Signed-off-by: Andrei Gherzan <andrei@balena.io>
---
 configs/rpi_4_32b_defconfig | 33 +++++++++++++++++++++++++++++++++
 configs/rpi_4_defconfig     | 33 +++++++++++++++++++++++++++++++++
 2 files changed, 66 insertions(+)
 create mode 100644 configs/rpi_4_32b_defconfig
 create mode 100644 configs/rpi_4_defconfig

diff --git a/configs/rpi_4_32b_defconfig b/configs/rpi_4_32b_defconfig
new file mode 100644
index 0000000000..a31a617a5f
--- /dev/null
+++ b/configs/rpi_4_32b_defconfig
@@ -0,0 +1,33 @@
+CONFIG_ARM=y
+CONFIG_ARCH_BCM283X=y
+CONFIG_SYS_TEXT_BASE=0x00008000
+CONFIG_TARGET_RPI_4_32B=y
+CONFIG_SYS_MALLOC_F_LEN=0x2000
+CONFIG_DISTRO_DEFAULTS=y
+CONFIG_NR_DRAM_BANKS=1
+# CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
+CONFIG_OF_BOARD=y
+CONFIG_OF_BOARD_SETUP=y
+CONFIG_MISC_INIT_R=y
+# CONFIG_DISPLAY_CPUINFO is not set
+# CONFIG_DISPLAY_BOARDINFO is not set
+CONFIG_SYS_PROMPT="U-Boot> "
+# CONFIG_CMD_FLASH is not set
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_FS_UUID=y
+CONFIG_ENV_FAT_INTERFACE="mmc"
+CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
+CONFIG_DM_KEYBOARD=y
+CONFIG_DM_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_BCM2835=y
+CONFIG_PINCTRL=y
+# CONFIG_PINCTRL_GENERIC is not set
+# CONFIG_REQUIRE_SERIAL_CONSOLE is not set
+CONFIG_DM_VIDEO=y
+CONFIG_SYS_WHITE_ON_BLACK=y
+CONFIG_CONSOLE_SCROLL_LINES=10
+CONFIG_PHYS_TO_BUS=y
+CONFIG_OF_LIBFDT_OVERLAY=y
diff --git a/configs/rpi_4_defconfig b/configs/rpi_4_defconfig
new file mode 100644
index 0000000000..da8c960a2a
--- /dev/null
+++ b/configs/rpi_4_defconfig
@@ -0,0 +1,33 @@
+CONFIG_ARM=y
+CONFIG_ARCH_BCM283X=y
+CONFIG_SYS_TEXT_BASE=0x00080000
+CONFIG_TARGET_RPI_4=y
+CONFIG_SYS_MALLOC_F_LEN=0x2000
+CONFIG_DISTRO_DEFAULTS=y
+CONFIG_NR_DRAM_BANKS=1
+# CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
+CONFIG_OF_BOARD=y
+CONFIG_OF_BOARD_SETUP=y
+CONFIG_MISC_INIT_R=y
+# CONFIG_DISPLAY_CPUINFO is not set
+# CONFIG_DISPLAY_BOARDINFO is not set
+CONFIG_SYS_PROMPT="U-Boot> "
+# CONFIG_CMD_FLASH is not set
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_FS_UUID=y
+CONFIG_ENV_FAT_INTERFACE="mmc"
+CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
+CONFIG_DM_KEYBOARD=y
+CONFIG_DM_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_BCM2835=y
+CONFIG_PINCTRL=y
+# CONFIG_PINCTRL_GENERIC is not set
+# CONFIG_REQUIRE_SERIAL_CONSOLE is not set
+CONFIG_DM_VIDEO=y
+CONFIG_SYS_WHITE_ON_BLACK=y
+CONFIG_CONSOLE_SCROLL_LINES=10
+CONFIG_PHYS_TO_BUS=y
+CONFIG_OF_LIBFDT_OVERLAY=y

From patchwork Wed Jul 24 14:39:05 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136357
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tygm6ldWz9s3l
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:40:36 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id 9DA33C21CB6; Wed, 24 Jul 2019 14:39:59 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-0.0 required=5.0 tests=RCVD_IN_MSPIKE_H2
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 17CAEC21D8E;
 Wed, 24 Jul 2019 14:39:47 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 82DD4C21CB6; Wed, 24 Jul 2019 14:39:36 +0000 (UTC)
Received: from mail-wr1-f67.google.com (mail-wr1-f67.google.com
 [209.85.221.67])
 by lists.denx.de (Postfix) with ESMTPS id 01AFFC21BE5
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:36 +0000 (UTC)
Received: by mail-wr1-f67.google.com with SMTP id 31so47328827wrm.1
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:36 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=SBRknuqaA8nrsEqGUKyxns0jOTIIj2fE4uzSYMyIGew=;
 b=PM1drqcqaap81ZOvnmAJY7X232GJ1xovWSwf0Gfii33S1JHlV0lXbcT6pHy/MY5C1Y
 GVqDd9jEF6nKvHAGx2xHd2df1KTAaYpuxqmmo89okoq1FQ80/gwzWj+AZJ/AsoZHiR0W
 Jo/QZmK8NjcG9xVEPkdvq2kYg7V7FGrTQgKFCn8y4tSk6rdFXH6x71luCFr9GTiHqmes
 XE5UGqaitR84yV9d5U4NL8pNnZbhtyxT8YYgkRyRbtOTwYi44C0XYsVwHP4G8DSjMjeE
 5jdjVUB8rmMki3Bqtj5aGkS0F+k/RvLiQPKbkssavqHl24M4ohWNlZ4fI9VXzg+4DJjZ
 aT7g==
X-Gm-Message-State: APjAAAXvh2LHWYIbEeX1TSDzEtnms+hdW6mVhDH8RRCsABLlzzPSCiUf
 8um70SOfA7+T6IFVzjiHE/2q21x6
X-Google-Smtp-Source: APXvYqzpQ+14M+bTmm58qDfrEm+q5psg5hqSSHCmXarzdL9SPFCPF5TlXBFq2NrJt8Zn5k7Cm+vgzQ==
X-Received: by 2002:adf:e444:: with SMTP id t4mr83713613wrm.262.1563979175247; 
 Wed, 24 Jul 2019 07:39:35 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.34
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:34 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:05 +0100
Message-Id: <20190724143911.29770-3-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 2/8] ARM: bcm283x: Add BCM283x_BASE define
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

From: Matthias Brugger <mbrugger@suse.com>

Devices of bcm283x have different base address, depending if they are on
bcm2835 or bcm2836/7. Use BCM283x_BASE depending on the SoC you want to
build and only add the offset in the header files.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
Signed-off-by: Andrei Gherzan <andrei@balena.io>
---
 arch/arm/mach-bcm283x/Kconfig              | 5 +++++
 arch/arm/mach-bcm283x/include/mach/mbox.h  | 6 +-----
 arch/arm/mach-bcm283x/include/mach/sdhci.h | 6 +-----
 arch/arm/mach-bcm283x/include/mach/timer.h | 6 +-----
 arch/arm/mach-bcm283x/include/mach/wdog.h  | 6 +-----
 5 files changed, 9 insertions(+), 20 deletions(-)

diff --git a/arch/arm/mach-bcm283x/Kconfig b/arch/arm/mach-bcm283x/Kconfig
index 3eb5a9a897..8e69914a83 100644
--- a/arch/arm/mach-bcm283x/Kconfig
+++ b/arch/arm/mach-bcm283x/Kconfig
@@ -141,4 +141,9 @@ config SYS_SOC
 config SYS_CONFIG_NAME
 	default "rpi"
 
+config BCM283x_BASE
+	hex
+	default "0x20000000" if BCM2835
+	default "0x3f000000" if BCM2836 || BCM2837
+
 endmenu
diff --git a/arch/arm/mach-bcm283x/include/mach/mbox.h b/arch/arm/mach-bcm283x/include/mach/mbox.h
index e3a893e49c..e44c7577da 100644
--- a/arch/arm/mach-bcm283x/include/mach/mbox.h
+++ b/arch/arm/mach-bcm283x/include/mach/mbox.h
@@ -37,11 +37,7 @@
 
 /* Raw mailbox HW */
 
-#ifndef CONFIG_BCM2835
-#define BCM2835_MBOX_PHYSADDR	0x3f00b880
-#else
-#define BCM2835_MBOX_PHYSADDR	0x2000b880
-#endif
+#define BCM2835_MBOX_PHYSADDR	(CONFIG_BCM283x_BASE + 0x0000b880)
 
 struct bcm2835_mbox_regs {
 	u32 read;
diff --git a/arch/arm/mach-bcm283x/include/mach/sdhci.h b/arch/arm/mach-bcm283x/include/mach/sdhci.h
index 5cb6ec3340..b443c379d8 100644
--- a/arch/arm/mach-bcm283x/include/mach/sdhci.h
+++ b/arch/arm/mach-bcm283x/include/mach/sdhci.h
@@ -6,11 +6,7 @@
 #ifndef _BCM2835_SDHCI_H_
 #define _BCM2835_SDHCI_H_
 
-#ifndef CONFIG_BCM2835
-#define BCM2835_SDHCI_BASE 0x3f300000
-#else
-#define BCM2835_SDHCI_BASE 0x20300000
-#endif
+#define BCM2835_SDHCI_BASE (CONFIG_BCM283x_BASE + 0x00300000)
 
 int bcm2835_sdhci_init(u32 regbase, u32 emmc_freq);
 
diff --git a/arch/arm/mach-bcm283x/include/mach/timer.h b/arch/arm/mach-bcm283x/include/mach/timer.h
index 56b0c356bb..014355e759 100644
--- a/arch/arm/mach-bcm283x/include/mach/timer.h
+++ b/arch/arm/mach-bcm283x/include/mach/timer.h
@@ -6,11 +6,7 @@
 #ifndef _BCM2835_TIMER_H
 #define _BCM2835_TIMER_H
 
-#ifndef CONFIG_BCM2835
-#define BCM2835_TIMER_PHYSADDR	0x3f003000
-#else
-#define BCM2835_TIMER_PHYSADDR	0x20003000
-#endif
+#define BCM2835_TIMER_PHYSADDR	(CONFIG_BCM283x_BASE + 0x00003000)
 
 #define BCM2835_TIMER_CS_M3	(1 << 3)
 #define BCM2835_TIMER_CS_M2	(1 << 2)
diff --git a/arch/arm/mach-bcm283x/include/mach/wdog.h b/arch/arm/mach-bcm283x/include/mach/wdog.h
index 99c88e5df7..8292b3cf1f 100644
--- a/arch/arm/mach-bcm283x/include/mach/wdog.h
+++ b/arch/arm/mach-bcm283x/include/mach/wdog.h
@@ -6,11 +6,7 @@
 #ifndef _BCM2835_WDOG_H
 #define _BCM2835_WDOG_H
 
-#ifndef CONFIG_BCM2835
-#define BCM2835_WDOG_PHYSADDR			0x3f100000
-#else
-#define BCM2835_WDOG_PHYSADDR			0x20100000
-#endif
+#define BCM2835_WDOG_PHYSADDR	(CONFIG_BCM283x_BASE + 0x00100000)
 
 struct bcm2835_wdog_regs {
 	u32 unknown0[7];

From patchwork Wed Jul 24 14:39:06 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136372
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tyl70FC2z9s3l
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:43:30 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id A8034C21DF3; Wed, 24 Jul 2019 14:40:46 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-0.0 required=5.0 tests=RCVD_IN_MSPIKE_H2
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 4C777C21DCA;
 Wed, 24 Jul 2019 14:40:10 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 474FDC21D74; Wed, 24 Jul 2019 14:39:39 +0000 (UTC)
Received: from mail-wm1-f68.google.com (mail-wm1-f68.google.com
 [209.85.128.68])
 by lists.denx.de (Postfix) with ESMTPS id 55C9DC21D4A
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:37 +0000 (UTC)
Received: by mail-wm1-f68.google.com with SMTP id s15so20507009wmj.3
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:37 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=pWvZ588J48xOgvZOVW9fhkWibsiRtNBVuM4VBynrLQw=;
 b=Am1Me0M/J3HedD2FzOBBvEpBhsNfG5psGW0Lap+uSDun9LynEo+RqAgmLAAp3KtUFD
 vUkieSMQZAjAIyzy8ySvUjuVA8iEUPN8wbo55HZ+3X5HfhgsbMwu70KZDA9UBw1T4v6V
 CE1pyL9enGgcwdDiTcXztZo57iM/vq1B6HilY8J27ZJFb9fJNH4L7IxE4O/febLJ4ljc
 micbK/014Ibg6DqlmHbBh4akENSDlXOrQLqMC8FdK0lsJIP7EH9Xqc6MubvgCzyhvS89
 xPFsmuRVAZVvpnQ6+mcSidfeiyYCMXh2IE8wgKExlyysSAjkZPC8Z+1H2qOnvOz+XEjX
 GVrg==
X-Gm-Message-State: APjAAAVcUgx+4UkyQKJDr3/tCu0zV2vBZk5reX2ukIrxHlkjQSBTYuSa
 Napy/FHrIjVsWObEdv0xTsrwwqT8
X-Google-Smtp-Source: APXvYqz1dOMBUWbTH/jK138AvmFcmCjdMc9s6oLE6NbSPeUiM4M0S3istSFU+IBlBRGqC2zlitzAQg==
X-Received: by 2002:a7b:cc86:: with SMTP id p6mr69114900wma.123.1563979176511; 
 Wed, 24 Jul 2019 07:39:36 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.35
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:35 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:06 +0100
Message-Id: <20190724143911.29770-4-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 3/8] ARM: bcm283x: Define configs for RaspberryPi 4
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

Define two target configs for Raspberry Pi 4 (32 and 64bit) and the
corresponding BCM2838* configs.

Be aware of the current limitation in firmware which requires an
explicit configuration to force the arm in 64bit mode when the
respective target is used.

Signed-off-by: Andrei Gherzan <andrei@balena.io>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 arch/arm/mach-bcm283x/Kconfig | 62 +++++++++++++++++++++++++++++++++++
 1 file changed, 62 insertions(+)

diff --git a/arch/arm/mach-bcm283x/Kconfig b/arch/arm/mach-bcm283x/Kconfig
index 8e69914a83..09a5b42bbb 100644
--- a/arch/arm/mach-bcm283x/Kconfig
+++ b/arch/arm/mach-bcm283x/Kconfig
@@ -26,6 +26,23 @@ config BCM2837_64B
 	select BCM2837
 	select ARM64
 
+config BCM2838
+	bool "Broadcom BCM2838 SoC support"
+	depends on ARCH_BCM283X
+
+config BCM2838_32B
+	bool "Broadcom BCM2838 SoC 32-bit support"
+	depends on ARCH_BCM283X
+	select BCM2838
+	select ARMV7_LPAE
+	select CPU_V7A
+
+config BCM2838_64B
+	bool "Broadcom BCM2838 SoC 64-bit support"
+	depends on ARCH_BCM283X
+	select BCM2838
+	select ARM64
+
 menu "Broadcom BCM283X family"
 	depends on ARCH_BCM283X
 
@@ -127,6 +144,50 @@ config TARGET_RPI_3
 	  This option creates a build targeting the ARMv8/AArch64 ISA.
 	select BCM2837_64B
 
+config TARGET_RPI_4_32B
+	bool "Raspberry Pi 4 32-bit build"
+	help
+	  Support for all BCM2838-based Raspberry Pi variants, such as
+	  the RPi 4 model B, in AArch32 (32-bit) mode.
+
+	  This option assumes the VideoCore firmware is configured to use the
+	  mini UART (rather than PL011) for the serial console. This is the
+	  default on the RPi 4. To enable the UART console, the following non-
+	  default option must be present in config.txt: enable_uart=1. This is
+	  required for U-Boot to operate correctly, even if you only care
+	  about the HDMI/usbkbd console.
+
+	  Due to hardware incompatibilities, this can't be used with
+	  BCM283/5/6/7.
+
+	  This option creates a build targeting the ARMv7/AArch32 ISA.
+	select BCM2838_32B
+
+config TARGET_RPI_4
+	bool "Raspberry Pi 4 64-bit build"
+	help
+	  Support for all BCM2838-based Raspberry Pi variants, such as
+	  the RPi 4 model B, in AArch64 (64-bit) mode.
+
+	  This option assumes the VideoCore firmware is configured to use the
+	  mini UART (rather than PL011) for the serial console. This is the
+	  default on the RPi 4. To enable the UART console, the following non-
+	  default option must be present in config.txt: enable_uart=1. This is
+	  required for U-Boot to operate correctly, even if you only care
+	  about the HDMI/usbkbd console.
+
+	  Due to hardware incompatibilities, this can't be used with
+	  BCM283/5/6/7.
+
+	  Also, due to a bug in firmware, switching to 64bit mode doesn't
+	  happen automatically based on the kernel's image filename. See
+	  https://github.com/raspberrypi/firmware/issues/1193 for more details.
+	  Until that is resolved, the configuration (config.txt) needs to
+	  explicitly set: arm_64bit=1.
+
+	  This option creates a build targeting the ARMv8/AArch64 ISA.
+	select BCM2838_64B
+
 endchoice
 
 config SYS_BOARD
@@ -145,5 +206,6 @@ config BCM283x_BASE
 	hex
 	default "0x20000000" if BCM2835
 	default "0x3f000000" if BCM2836 || BCM2837
+	default "0xfe000000" if BCM2838
 
 endmenu

From patchwork Wed Jul 24 14:39:07 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136366
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tyjX1rBvz9s3l
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:42:08 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id F1D99C21CB6; Wed, 24 Jul 2019 14:40:11 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-0.0 required=5.0 tests=RCVD_IN_MSPIKE_H2
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 937A0C21D65;
 Wed, 24 Jul 2019 14:39:51 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 748BCC21C2F; Wed, 24 Jul 2019 14:39:40 +0000 (UTC)
Received: from mail-wm1-f68.google.com (mail-wm1-f68.google.com
 [209.85.128.68])
 by lists.denx.de (Postfix) with ESMTPS id CE175C21C29
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:37 +0000 (UTC)
Received: by mail-wm1-f68.google.com with SMTP id a15so42072601wmj.5
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:37 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=u/X0NAp+r55Bugx2HDgeaVfXBeYCSiyWcV7zL0w2yXE=;
 b=meBOAW1SzAbuPxzp+nYt716FpccdfeH1Xh3xWr+zMzkGW0O6P+hkYAjOsYmnzj/5rX
 WNazbP1wNvSVBowPChoHa1zwn2I8PhxW9EDSfNkECgfLCYdPettoGRQ94zsUSZAkp4dD
 FJkwXFsPDQX4UoFpLxN7fhbkYqdcezpK7kv2ijiYnQR9z/RfadD4yVQHLYt+M6UsmWph
 FjXmBWf8Dqj0HniyCpAFBQclBp2LF6Cf3teLv0SKjJfkyJnfnB2D26Sig1t5Z1JIRw8h
 +2J1FXpChqug7KpnKuQNHWz1P/yhSY/b+dPHkLABO1yCQ9N5+IvUCw+KXXd8o3RtpwuS
 UnjA==
X-Gm-Message-State: APjAAAULyjSTt+VemoBBXMausXImPmWbP6ymxMkyPmdzjRXNS/pvDSjw
 SvFvIJXwqw2UxgfJbVQsZf3L04kE
X-Google-Smtp-Source: APXvYqwDVdWVyjv8DnkgBTqEwcrcxXXF1Xof6TjgZnG7RMtXzWC3l2a8f4Awp1bVLNcspHnduMBWqg==
X-Received: by 2002:a1c:a909:: with SMTP id s9mr74064408wme.20.1563979177108; 
 Wed, 24 Jul 2019 07:39:37 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.36
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:36 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:07 +0100
Message-Id: <20190724143911.29770-5-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 4/8] RPI: Add entry for Raspberry Pi 4 model B
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

The Raspebrry Pi 4 uses the new revision code scheme as documented by
the foundation. This change adds an entry for this board as well.

Signed-off-by: Andrei Gherzan <andrei@balena.io>
---
 board/raspberrypi/rpi/rpi.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/board/raspberrypi/rpi/rpi.c b/board/raspberrypi/rpi/rpi.c
index 617c892dde..6d6f1ef39a 100644
--- a/board/raspberrypi/rpi/rpi.c
+++ b/board/raspberrypi/rpi/rpi.c
@@ -148,6 +148,11 @@ static const struct rpi_model rpi_models_new_scheme[] = {
 		DTB_DIR "bcm2837-rpi-cm3.dtb",
 		false,
 	},
+	[0x11] = {
+		"4 Model B",
+		DTB_DIR "bcm2711-rpi-4-b.dtb",
+		true,
+	},
 };
 
 static const struct rpi_model rpi_models_old_scheme[] = {

From patchwork Wed Jul 24 14:39:08 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136373
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tyl91yqVz9s3l
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:43:33 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id 0991BC21D74; Wed, 24 Jul 2019 14:40:59 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=none autolearn=unavailable
 autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 118D3C21D8E;
 Wed, 24 Jul 2019 14:40:25 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 3020BC21DA2; Wed, 24 Jul 2019 14:39:40 +0000 (UTC)
Received: from mail-wr1-f65.google.com (mail-wr1-f65.google.com
 [209.85.221.65])
 by lists.denx.de (Postfix) with ESMTPS id 9468EC21C4A
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:38 +0000 (UTC)
Received: by mail-wr1-f65.google.com with SMTP id r1so47270575wrl.7
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:38 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=INTj2ZkmrgCvQnwXSkRkNgGGyv8wKd+4Dp1WiNdZg9A=;
 b=Q12tNYAAjgCy4zkzIOEnY6VGsX14oy+bwAD75RZPvG/KwLZEgoIWaH67glrnxVMKCW
 lnIUiim0jTu+lUQws/OaHw3Nl55HUaXoZMrKZ50vANqXzk0brozRNLBsZTBmFZGlyrP4
 1ftizkEJMJr74KxyuZBbSHmN5gMxoEH/r6V8DZ1b8dH5RXr0J4hEuoeDiKT26fxfOpMk
 dLXJd/Mx/4t6AVkcdVJIs3jdcw4neuiDThpHeLeNiuZgKEIOy7HHs+Zfz4iTRmu5IrHK
 LwIYdSl6UeGnJNPE595PyB1HtUzUwEoUWuyzP062TGzGxfEvMjnFJ5bzm/dV9sOSNBtx
 308g==
X-Gm-Message-State: APjAAAUR5WDXsRNKCCMKSHatakaD4J0JzmZMzF3dOkiIj5/vqS01j3nF
 6aVr6nFQUuBmYJrMYEjaAwJqc6dm
X-Google-Smtp-Source: APXvYqwd9TOCUBNszoD7ZZhvAYwth50GqOLaqpKiC/HlGaXndFYzgPMhMhFr3nL9Gf6MfsBZ5Sho3Q==
X-Received: by 2002:adf:eb0f:: with SMTP id
 s15mr12285481wrn.324.1563979177934; 
 Wed, 24 Jul 2019 07:39:37 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.37
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:37 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:08 +0100
Message-Id: <20190724143911.29770-6-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 5/8] ARM: bcm283x: Include definition for
 additional emmc clock
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

This clock has a different mbox ID so have this included in the relevant
header file.

Signed-off-by: Andrei Gherzan <andrei@balena.io>
---
 arch/arm/mach-bcm283x/include/mach/mbox.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-bcm283x/include/mach/mbox.h b/arch/arm/mach-bcm283x/include/mach/mbox.h
index e44c7577da..f2a98acddd 100644
--- a/arch/arm/mach-bcm283x/include/mach/mbox.h
+++ b/arch/arm/mach-bcm283x/include/mach/mbox.h
@@ -230,6 +230,7 @@ struct bcm2835_mbox_tag_set_power_state {
 #define BCM2835_MBOX_CLOCK_ID_SDRAM	8
 #define BCM2835_MBOX_CLOCK_ID_PIXEL	9
 #define BCM2835_MBOX_CLOCK_ID_PWM	10
+#define BCM2835_MBOX_CLOCK_ID_EMMC2	12
 
 struct bcm2835_mbox_tag_get_clock_rate {
 	struct bcm2835_mbox_tag_hdr tag_hdr;

From patchwork Wed Jul 24 14:39:09 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136365
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tyjV5JXbz9sBF
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:42:06 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id 77213C21C27; Wed, 24 Jul 2019 14:40:36 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-0.0 required=5.0 tests=RCVD_IN_MSPIKE_H2
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id A24EDC21DB5;
 Wed, 24 Jul 2019 14:40:00 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 4193AC21C2F; Wed, 24 Jul 2019 14:39:42 +0000 (UTC)
Received: from mail-wr1-f68.google.com (mail-wr1-f68.google.com
 [209.85.221.68])
 by lists.denx.de (Postfix) with ESMTPS id AE825C21D4A
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:39 +0000 (UTC)
Received: by mail-wr1-f68.google.com with SMTP id 31so47329072wrm.1
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:39 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=fpnN0uOMvR4DjRJfxVux4tdvICr5Rjygmvsk8TlIvnA=;
 b=MQVZ2YY2EKhWHLLaavQarKM51ls7jOHZ7N8mWZpNRCJNQtRJfYkVDlHw5vbIV2dFXF
 W6qAp6fnL+fb9/RHiw786uJCu/L1CZ95P9ksjyGYE62VQLWFE5OxDF2NGkoxZ0HGkqiw
 cMAcpX5+Zn+/ZA6Q7OGOz1dZqASEH8mWaPI5TS1afkxMn5E14RKVHN6//bkvXwIpXL7d
 0M7zjoEQGuDlZK3PoRhxESbXrtznvm2PS2xiTdgtH5i9dJ24bbfjQulbBx9EMR++tB8O
 e+SLxw9eT3N8XZwF0YD4XJRcFc8kwyiwhKFU2tuNJTC7pIkBSOeLxUUyJGsypxREq54n
 11GA==
X-Gm-Message-State: APjAAAXU6kbWgi4nhW5lhQR6pbZIhBdAkcKNmC1i59gWXdb8lL6dbHqL
 Q2Ye3JGmLehlU2z4+7T/uXzHhO+V
X-Google-Smtp-Source: APXvYqyYzkP4kraoV8icuHlxBOcusLrizbOSQSILHq90ieWqGbICGdj7ZQRO5lX7n4DS1rr/nKThUQ==
X-Received: by 2002:adf:ea82:: with SMTP id s2mr82009745wrm.91.1563979179031; 
 Wed, 24 Jul 2019 07:39:39 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.38
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:38 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:09 +0100
Message-Id: <20190724143911.29770-7-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 6/8] mmc: bcm283x: Add support for bcm2711
 device in bcm2835_sdhci
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

From: Matthias Brugger <mbrugger@suse.com>

The bcm2711 has two emmc controllers. The difference is the clocks
they use. Add support for the second emmc controller.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
Signed-off-by: Andrei Gherzan <andrei@balena.io>
---
 drivers/mmc/bcm2835_sdhci.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/bcm2835_sdhci.c b/drivers/mmc/bcm2835_sdhci.c
index 08bddd410e..e68dec3be7 100644
--- a/drivers/mmc/bcm2835_sdhci.c
+++ b/drivers/mmc/bcm2835_sdhci.c
@@ -178,12 +178,13 @@ static int bcm2835_sdhci_probe(struct udevice *dev)
 	fdt_addr_t base;
 	int emmc_freq;
 	int ret;
+	int clock_id = (int)dev_get_driver_data(dev);
 
 	base = devfdt_get_addr(dev);
 	if (base == FDT_ADDR_T_NONE)
 		return -EINVAL;
 
-	ret = bcm2835_get_mmc_clock(BCM2835_MBOX_CLOCK_ID_EMMC);
+	ret = bcm2835_get_mmc_clock(clock_id);
 	if (ret < 0) {
 		debug("%s: Failed to set MMC clock (err=%d)\n", __func__, ret);
 		return ret;
@@ -228,7 +229,14 @@ static int bcm2835_sdhci_probe(struct udevice *dev)
 }
 
 static const struct udevice_id bcm2835_sdhci_match[] = {
-	{ .compatible = "brcm,bcm2835-sdhci" },
+	{
+		.compatible = "brcm,bcm2835-sdhci",
+		.data = BCM2835_MBOX_CLOCK_ID_EMMC
+	},
+	{
+		.compatible = "brcm,bcm2711-emmc2",
+		.data = BCM2835_MBOX_CLOCK_ID_EMMC2
+	},
 	{ /* sentinel */ }
 };
 

From patchwork Wed Jul 24 14:39:10 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136369
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tyk04V3zz9s3l
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:42:32 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id 47E81C21DD3; Wed, 24 Jul 2019 14:41:08 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-0.0 required=5.0 tests=RCVD_IN_MSPIKE_H2
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id A7B62C21C93;
 Wed, 24 Jul 2019 14:40:30 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 05EE7C21D65; Wed, 24 Jul 2019 14:39:43 +0000 (UTC)
Received: from mail-wr1-f67.google.com (mail-wr1-f67.google.com
 [209.85.221.67])
 by lists.denx.de (Postfix) with ESMTPS id A0C53C21CB1
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:40 +0000 (UTC)
Received: by mail-wr1-f67.google.com with SMTP id p17so47250123wrf.11
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:40 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=vke+aeJLlm+hH0ZbApV2tQXwcQSNthXhQrHw3uTUjCY=;
 b=fXeMysXcz5ZFzv+beEbqCcjHVfBZuIIVE7p40QdU2ckI7M3z1jdwMCXENGpwnVdMks
 H+k4EDysfUSZmD9YGLyX+rFUMPgHEZm7DcxpQGDoT28zXRKo1I/5BQhEm5VHZYudmzLM
 A9Juze34OYZ+qop/80ZSAARNy/grkz+9EL0hz+/cGoKbQM2vnYeK9IhqY1MT+Jnjaqeo
 Avmh6DcXZN5MtJOGGt7TsXETUzeCZViAJDpgT+lIy8aFAABW4ZHBF6eZtgzd/aNn0dk7
 RJHyQlXxpKIeZGVzZz6kO+VFwHV++awYsZK1wzhqNDvAA0kGbkvgIWc1qOii+4j351tr
 FmfA==
X-Gm-Message-State: APjAAAUddD7/P+xcAGZeEeanF39Kb1CjavSwOjql/I604nDwgzGThPdu
 QltkJmJtpo31uSwBMKkalJwi1Xg3
X-Google-Smtp-Source: APXvYqydROkNjRTwQxNzaMfPcjFShYCgunskcjaWkRPeJsf9HRFmaH/uThvtF9xRw5cFaHtr2ixI2w==
X-Received: by 2002:a5d:668e:: with SMTP id
 l14mr31747640wru.156.1563979179920; 
 Wed, 24 Jul 2019 07:39:39 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.39
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:39 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:10 +0100
Message-Id: <20190724143911.29770-8-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 7/8] RPI: Add memory map for bcm2838
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

Define the memory map for the BCM2838 based on the dt configuration
available in the Raspberry Pi kernel fork.

Signed-off-by: Andrei Gherzan <andrei@balena.io>
---
 board/raspberrypi/rpi/rpi.c | 27 ++++++++++++++++++++++++---
 1 file changed, 24 insertions(+), 3 deletions(-)

diff --git a/board/raspberrypi/rpi/rpi.c b/board/raspberrypi/rpi/rpi.c
index 6d6f1ef39a..1c4fae9166 100644
--- a/board/raspberrypi/rpi/rpi.c
+++ b/board/raspberrypi/rpi/rpi.c
@@ -249,7 +249,8 @@ static uint32_t rev_type;
 static const struct rpi_model *model;
 
 #ifdef CONFIG_ARM64
-static struct mm_region bcm2837_mem_map[] = {
+#ifndef CONFIG_BCM2838
+static struct mm_region bcm283x_mem_map[] = {
 	{
 		.virt = 0x00000000UL,
 		.phys = 0x00000000UL,
@@ -268,8 +269,28 @@ static struct mm_region bcm2837_mem_map[] = {
 		0,
 	}
 };
-
-struct mm_region *mem_map = bcm2837_mem_map;
+#else
+static struct mm_region bcm283x_mem_map[] = {
+	{
+		.virt = 0x00000000UL,
+		.phys = 0x00000000UL,
+		.size = 0xfe000000UL,
+		.attrs = PTE_BLOCK_MEMTYPE(MT_NORMAL) |
+			 PTE_BLOCK_INNER_SHARE
+	}, {
+		.virt = 0xfe000000UL,
+		.phys = 0xfe000000UL,
+		.size = 0x01800000UL,
+		.attrs = PTE_BLOCK_MEMTYPE(MT_DEVICE_NGNRNE) |
+			 PTE_BLOCK_NON_SHARE |
+			 PTE_BLOCK_PXN | PTE_BLOCK_UXN
+	}, {
+		/* List terminator */
+		0,
+	}
+};
+#endif
+struct mm_region *mem_map = bcm283x_mem_map;
 #endif
 
 int dram_init(void)

From patchwork Wed Jul 24 14:39:11 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrei Gherzan <andrei@balena.io>
X-Patchwork-Id: 1136370
X-Patchwork-Delegate: matthias.bgg@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=balena.io
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 45tyk61mBqz9s3l
 for <incoming@patchwork.ozlabs.org>;
 Thu, 25 Jul 2019 00:42:38 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id EF268C21DA2; Wed, 24 Jul 2019 14:40:21 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=none autolearn=unavailable
 autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 18B41C21DDC;
 Wed, 24 Jul 2019 14:39:52 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id DD62DC21C2F; Wed, 24 Jul 2019 14:39:44 +0000 (UTC)
Received: from mail-wm1-f66.google.com (mail-wm1-f66.google.com
 [209.85.128.66])
 by lists.denx.de (Postfix) with ESMTPS id 8735EC21DB3
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 14:39:41 +0000 (UTC)
Received: by mail-wm1-f66.google.com with SMTP id 207so42087264wma.1
 for <u-boot@lists.denx.de>; Wed, 24 Jul 2019 07:39:41 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=Su0sVvS3yD62leuIKZb3pxg8i/NN5tZXXSx8RoqiHBw=;
 b=d7yjVaKAs5z1TFruY3Dqyo5v+F+4E+6cfpsthOMt+m9MZeGD4gERwCAslKdO4akkyp
 wWRBoQCMMMWzwpoRBvjczC5hM7xQ/pKqEZAJ9KP5gyzmlCmx+CznVjJGkVsbKE/9Mjz8
 8fTflQbyCszmiwMdPOjMJnLHi5cbT1R0qbiwMeYPNkab9mmWczX7x6u9JfzS51kG3G3A
 hD5QehF28N7XODXttdFd1kMGzEC+6xzJPdYrCcGCoGeOBkg3KiHS/dkB8sXp/tvwHQv/
 4dPZmjtq5exSdZ0Mm2E5RsvISeMK6G8Iqv0PcUXUGYBQryXiw1DDve4wnDR1gw/YJNV0
 07yQ==
X-Gm-Message-State: APjAAAVxgFnLbdj7oxAR7jyH5MO8StfPsSq/KejdOEq2tDuha3i9EwGn
 dS6MsHWnNMhYVuzQdHZwaQ2aOYon
X-Google-Smtp-Source: APXvYqyVLBWemfnQgqJJ6fX0KGAaRRkuU+DwxqsmiT4cxbq4NFvAxBdWRSZ2PrRk7ZhY55qr3+jwcg==
X-Received: by 2002:a1c:a997:: with SMTP id
 s145mr72938337wme.106.1563979180848; 
 Wed, 24 Jul 2019 07:39:40 -0700 (PDT)
Received: from localhost.localdomain ([212.36.34.46])
 by smtp.gmail.com with ESMTPSA id
 n14sm84860651wra.75.2019.07.24.07.39.40
 (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
 Wed, 24 Jul 2019 07:39:40 -0700 (PDT)
From: Andrei Gherzan <andrei@balena.io>
To: u-boot@lists.denx.de
Date: Wed, 24 Jul 2019 15:39:11 +0100
Message-Id: <20190724143911.29770-9-andrei@balena.io>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190724143911.29770-1-andrei@balena.io>
References: <20190716133803.1174-1-andrei@gherzan.ro>
 <20190724143911.29770-1-andrei@balena.io>
MIME-Version: 1.0
Cc: mbrugger@suse.com, Andrei Gherzan <andrei@balena.io>
Subject: [U-Boot] [PATCH v2 8/8] git-mailrc: Add rpi and bcm283x maintainer
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

Signed-off-by: Andrei Gherzan <andrei@balena.io>
Acked-by: Matthias Brugger <mbrugger@suse.com>
---
 doc/git-mailrc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/doc/git-mailrc b/doc/git-mailrc
index a63b76befc..68110e1963 100644
--- a/doc/git-mailrc
+++ b/doc/git-mailrc
@@ -35,6 +35,7 @@ alias mariosix       Mario Six <mario.six@gdsys.cc>
 alias masahiro       Masahiro Yamada <yamada.masahiro@socionext.com>
 alias mateusz        Mateusz Kulikowski <mateusz.kulikowski@gmail.com>
 alias maxime         Maxime Ripard <maxime.ripard@free-electrons.com>
+alias mbrugger       Matthias Brugger <mbrugger@suse.com>
 alias monstr         Michal Simek <monstr@monstr.eu>
 alias prom           Minkyu Kang <mk7.kang@samsung.com>
 alias ptomsich       Philipp Tomsich <philipp.tomsich@theobroma-systems.com>
@@ -74,6 +75,9 @@ alias uniphier       uboot, masahiro
 alias zynq           uboot, monstr
 alias rockchip       uboot, sjg, kevery, ptomsich
 
+alias bcm283x        uboot,mbrugger
+alias rpi            uboot,mbrugger
+
 alias m68k           uboot, alisonwang, angelo_ts
 alias coldfire       m68k
 
From 970baf16d1322d3930a57fc78ddfb15d594d690c Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fvogt@suse.com>
Date: Thu, 11 Jul 2019 16:56:24 +0200
Subject: [PATCH] video: arm: rpi: Bail out early if querying video information
 fails

When probing we query for the width and hight of the display. If the
firmware does not report any connected display the system will crash.
See https://github.com/raspberrypi/firmware/issues/1157 for details.

Signed-off-by: Fabian Vogt <fvogt@suse.com>
[mb: update commit message]
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
Reviewed-by: Andre Przywara <andre.przywara@arm.com>
Tested-by: Andre Przywara <andre.przywara@arm.com>

diff --git a/drivers/video/bcm2835.c b/drivers/video/bcm2835.c
index bc41090aed..1d2eda084c 100644
--- a/drivers/video/bcm2835.c
+++ b/drivers/video/bcm2835.c
@@ -19,13 +19,15 @@ static int bcm2835_video_probe(struct udevice *dev)
 
 	debug("bcm2835: Query resolution...\n");
 	ret = bcm2835_get_video_size(&w, &h);
-	if (ret)
+	if (ret || w == 0 || h == 0)
 		return -EIO;
 
 	debug("bcm2835: Setting up display for %d x %d\n", w, h);
 	ret = bcm2835_set_video_params(&w, &h, 32, BCM2835_MBOX_PIXEL_ORDER_RGB,
 				       BCM2835_MBOX_ALPHA_MODE_IGNORED,
 				       &fb_base, &fb_size, &pitch);
+	if (ret)
+		return -EIO;
 
 	debug("bcm2835: Final resolution is %d x %d\n", w, h);
 
